pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -60,13 +60,9 @@ func syncOpenShiftControllerManager_v311_00_to_latest(
 	operandName := "openshift-controller-manager"
 	rcOperandName := "route-controller-manager"
 
-	specAnnotations := map[string]string{
-		"openshiftcontrollermanagers.operator.openshift.io/cluster": strconv.FormatInt(operatorConfig.ObjectMeta.Generation, 10),
```

Comment:
the changes look ok to me at the first look, but can you test that the operand is getting updated on the rest of changes of OpenShiftControllerManager object?
- logLevel
- observedConfig
- unsupportedConfigOverrides

Reply:
I have tested the 3 cases u mentioned above, it all worked as expected, the results are shown as follows:

1. change the logvelvel from Normal to Debug, the operator logs show as 
```
I0315 02:43:16.774303       1 event.go:285] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-controller-manager-operator", Name:"openshift-controller-manager-operator", UID:"caaaf76d-d550-4cc7-a596-4eace4439a64", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'DeploymentUpdated' Updated Deployment.apps/controller-manager -n openshift-controller-manager because it changed
I0315 02:43:16.792360       1 event.go:285] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-controller-manager-operator", Name:"openshift-controller-manager-operator", UID:"caaaf76d-d550-4cc7-a596-4eace4439a64", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Status for clusteroperator/openshift-controller-manager changed: Progressing changed from False to True ("Progressing: deployment/controller-manager: observed generation is 10, desired generation is 11.\nProgressing: openshiftcontrollermanagers.operator.openshift.io/cluster: observed generation is 12, desired generation is 13.")
```

2. change the network cr  sepc.externalIP.autoAssignCIDRs, the operator logs show as:
```
I0315 02:40:36.765820       1 event.go:285] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-controller-manager-operator", Name:"openshift-controller-manager-operator", UID:"caaaf76d-d550-4cc7-a596-4eace4439a64", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ConfigMapUpdated' Updated ConfigMap/config -n openshift-controller-manager:
cause by changes in data.config.yaml
I0315 02:40:36.800724       1 event.go:285] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-controller-manager-operator", Name:"openshift-controller-manager-operator", UID:"caaaf76d-d550-4cc7-a596-4eace4439a64", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Status for clusteroperator/openshift-controller-manager changed: Progressing changed from False to True ("Progressing: deployment/controller-manager: observed generation is 9, desired generation is 10.\nProgressing: openshiftcontrollermanagers.operator.openshift.io/cluster: observed generation is 11, desired generation is 12.")


```
3. change the unsupportedConfigOverrides with 
```
  unsupportedConfigOverrides:
    imageImport:
      MaxScheduledImageImportsPerMinute: 50
```
and the operator log show as :
```
I0315 02:40:36.765820       1 event.go:285] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-controller-manager-operator", Name:"openshift-controller-manager-operator", UID:"caaaf76d-d550-4cc7-a596-4eace4439a64", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ConfigMapUpdated' Updated ConfigMap/config -n openshift-controller-manager:
cause by changes in data.config.yaml
I0315 02:40:36.800724       1 event.go:285] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-controller-manager-operator", Name:"openshift-controller-manager-operator", UID:"caaaf76d-d550-4cc7-a596-4eace4439a64", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Status for clusteroperator/openshift-controller-manager changed: Progressing changed from False to True ("Progressing: deployment/controller-manager: observed generation is 9, desired generation is 10.\nProgressing: openshiftcontrollermanagers.operator.openshift.io/cluster: observed generation is 11, desired generation is 12.")

```

Reply:
How's the logic of not rendering a new rollout based on operatorloglevel related to removing the annotations? The relation is not clear through the code change.

Reply:
> How's the logic of not rendering a new rollout based on operatorloglevel related to removing the annotations? The relation is not clear through the code change. Would you please update the PR description mentioning how this works?

the PR message and git log message has updated. please check again, thx

Reply:
I think the PR and commit message are not sufficient - we need a code comment explaining why we shouldn't track the operator config generation _at all_. How will changes to the Build and Image cluster config objects get propagated to openshift-controller-manager? This is not explained.

I'm also concerned that this will break our Capabilities for Build and DeploymentConfig, which does not have good end to end testing.

Reply:
yeah, feel free to add e2e tests for this